types/actions-on-google/index.d.ts

```diff
@@ -0,0 +1,39 @@
+// Type definitions for actions-on-google 1.4
+// Project: https://github.com/actions-on-google/actions-on-google-nodejs
+// Definitions by: Joel Hegg <https://github.com/joelhegg>, Pilwon Huh <https://github.com/pilwon>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Looks OK, except that I'm not sure whether the Apache 2.0 License declaration conflicts with the repository's MIT license.

Reply:
This isn't the first time this project has run into this. Some examples of existing type definitions tagged with an Apache license: applepayjs, bootstrap-switch, core-js, date.format.js, es6-collections, google-apps-script, highland, winrt.

Reply:
Take `@types/core-js` as example, I `npm install` it, I can find a `LICENSE` file of `MIT License`, and `"license": "MIT"` in `package.json` file, but the code has an Apache 2.0 License declaration, it confuses me what its license is.
@andy-ms

Reply:
Hi, 

the Apache 2.0 License is the license on which the project is published. But the Apache 2.0 License allows you to publish own software under your license, when you pointed out that the original software is published under Apache 2.0, which is already present in the header of each of the file.

So @joelhegg is it possible, to add the MIT License to the root of the project and in the package.json file? 
This would be the license of your own work and this commit.

Reply:
Those should not have been merged. Different licenses are a very big problem for us because we automated tasks that depend on the license being an appropriate one in order for the tool's behavior to be legal.

I'm not super concerned about Apache (seems like it's OK but I haven't cleared it with our legal team) but unless there's a very, very good reason to introduce more problems here, I can't merge this.

Reply:
So it sounds like the options are:
1. @RyanCavanaugh to clear the usage of the Apache license within type definitions with his legal folks.
1. Remove the license headers as suggested by @andy-ms and add a NOTICE similar to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google-apps-script/NOTICE
1. Use dts-gen to generate a clean type definition, but lose several thousand lines of documentation.
1. Abandon the PR.

Please let me know how you'd like to proceed.

Reply:
@plantain-00 @Thomas-P @andy-ms @RyanCavanaugh Based on the following response from Google in actions-on-google/actions-on-google-nodejs#2 I removed the license headers and added a NOTICE. Please let me know if this addresses your concerns.

> @joelhegg from what I've seen the pull request that you have submitted doesn't contain any files authored by us, so we have no claims on them. You can strip the headers and change the license, although you should probably check with @pilwon as it looks like your work is based off what he's done. For the jsdoc comments, as you mentioned you can attribute them to our docs released under CC, adding a NOTICE similar to the one you point to in your comments.
>
> As @Thomas-P commented on the DefinitelyTyped pull request, in a repo like @pilwon's fork where you have some files authored by us and some not, you can change the license for the files you authored, you'd only need to keep our headers in the file authored by us, a copy of the apache license in the repo (as per point 4a of the license), notices for files authored by us that you have changed.